### ISSUE TITLE
The checks `LocalFunctionDecompiler.IsLocalFunctionMethod()` from `CSharpDecompiler.ReadCodeMappingInfo()` may be too expensive?

### ISSUE THREAD

**Issue by yyjdelete** [icsharpcode/ILSpy#1672]
Tested with 5.0.0.5097-preview4

Seems it will be call 6970 times when decompile an single class(`ICSharpCode.Decompiler.CSharp.CSharpDecompiler`), and will take 10~20% total time of decompile an assembly with Release builds of ILSpy(without CheckInvariant).

https://github.com/icsharpcode/ILSpy/blob/9358ea6d7381daaae155eb315d84ba1c0b7831b8/ICSharpCode.Decompiler/CSharp/CSharpDecompiler.cs#L712
```cs
at ICSharpCode.Decompiler.IL.Transforms.LocalFunctionDecompiler.IsLocalFunctionMethod(PEFile module, MethodDefinitionHandle methodHandle, ILTransformContext context)
at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.ReadCodeMappingInfo(PEFile module, CodeMappingInfo info, MethodDefinitionHandle parent, MethodDefinitionHandle part, Queue`1 connectedMethods, HashSet`1 processedNestedTypes)
at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.GetCodeMappingInfo(PEFile module, EntityHandle member)
at ICSharpCode.Decompiler.CSharp.RequiredNamespaceCollector.CollectNamespaces(IEntity entity, MetadataModule module, CodeMappingInfo mappingInfo)
at ICSharpCode.Decompiler.CSharp.RequiredNamespaceCollector.CollectNamespaces(IEntity entity, MetadataModule module, HashSet`1 namespaces)
at ICSharpCode.Decompiler.CSharp.RequiredNamespaceCollector.CollectNamespaces(EntityHandle entity, MetadataModule module, HashSet`1 namespaces)
at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.Decompile(IEnumerable`1 definitions)
at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.Decompile(EntityHandle[] definitions)
at ICSharpCode.ILSpy.CSharpLanguage.DecompileType(ITypeDefinition type, ITextOutput output, DecompilationOptions options)
at ICSharpCode.ILSpy.TreeNodes.TypeTreeNode.Decompile(Language language, ITextOutput output, DecompilationOptions options)
at ICSharpCode.ILSpy.TextView.DecompilerTextView.DecompileNodes(DecompilationContext context, ITextOutput textOutput)
at ICSharpCode.ILSpy.TextView.DecompilerTextView.<>c__DisplayClass36_0.<DecompileAsync>b__0()
```

**Comment by siegfriedpammer** [comment]
@yyjdelete was this fixed by #1763 ?

**Comment by yyjdelete** [comment]
@siegfriedpammer 
Yes. Verified it's also fixed with #1763. I didn't notice that they have almost the same stacktrace.
